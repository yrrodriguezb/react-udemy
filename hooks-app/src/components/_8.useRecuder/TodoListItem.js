import React from 'react'
import PropTypes from 'prop-types';

export const TodoListItem = ({ todo, index, handleToggle,  handleDelete }) => {
  return (
    <li
      key={ todo.id }
      className='d-flex justify-content-between list-group-item'
    >
      <span 
        className={ `${todo.done && 'text-decoration-line-through'}` }
        onClick={ () => { handleToggle( todo.id ) } }
      >
        { index + 1 }. { todo.desc }
      </span>
      <input  
        className="btn btn-danger btn-sm" 
        type="button" 
        value="Borrar" 
        onClick={ () => handleDelete( todo.id ) }
        />
    </li>
  )
}

TodoListItem.propTypes = {
  todo: PropTypes.object,
  index: PropTypes.number,
  handleToggle: PropTypes.func,
  handleDelete: PropTypes.func
}
