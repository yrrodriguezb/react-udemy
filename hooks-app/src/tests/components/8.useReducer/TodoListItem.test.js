import React from 'react'
import { shallow } from 'enzyme'
import { TodoListItem } from '../../../components/_8.useRecuder/TodoListItem'
import { demoTodos } from '../../fixtures/demoTodos'
import '@testing-library/jest-dom'


describe('Pruebas en <TodoListItem />', () => {
  const todo = demoTodos[0]

  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow(
    <TodoListItem 
      todo={ todo }
      index={ 0 }
      handleDelete={ handleDelete }
      handleToggle={ handleToggle }
    />
  )
  
  test('Debe mostrarse correctamente', async () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('Debe llamar la funcion handleDelete ', async () => {
    wrapper.find('input').simulate('click');
    expect(handleDelete).toHaveBeenCalledWith(todo.id)
  });

  test('Debe llamar la funcion handleToggle ', async () => {
    wrapper.find('span').simulate('click');
    expect(handleToggle).toHaveBeenCalledWith(todo.id)
  })

  test('Debe mostrar el texto correctamente', async () => {
    const span = wrapper.find("span");
    const text = `1. ${todo.desc}`
    expect(wrapper.text().trim()).toBe(text)
  })

  test('Debe tener la clase "text-decoration-line-through" si el todo esta en "done"', async () => {
    todo.done = true;

    const wrapper = shallow(
      <TodoListItem 
        todo={ todo }
        index={ 0 }
        handleDelete={ handleDelete }
        handleToggle={ handleToggle }
      />
    )

    expect(wrapper.find("span").hasClass('text-decoration-line-through')).toBe(true);
  });
})