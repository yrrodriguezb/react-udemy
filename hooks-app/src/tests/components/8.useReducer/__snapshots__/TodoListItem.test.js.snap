// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoListItem /> Debe mostrarse correctamente 1`] = `
<li
  className="d-flex justify-content-between list-group-item"
  key="1"
>
  <span
    className="false"
    onClick={[Function]}
  >
    1
    . 
    Aprender React
  </span>
  <input
    className="btn btn-danger btn-sm"
    onClick={[Function]}
    type="button"
    value="Borrar"
  />
</li>
`;
